### DISCUSSION TITLE
有人使用过平台插件吗？

### DISCUSSION THREAD

**Discussion by unclechong** [NervJS/taro#13229] [Q & A] [answered]
只能定义新的平台插件，不能基于现有的平台插件做拓展吗？

**Comment by tourze** [comment]
理论上，qq就是基于weapp平台插件来实现的，copy qq那个package出来改改咯

**Comment by AdvancedCat** [comment, marked as answer]
横向扩展：新起一个平台插件，以支持一个新厂商小程序，如 `taro-weapp`，目前 Taro 主仓内置了常见的几个平台插件；

纵向扩展：厂商内部多个业务类小程序，比如支付宝小程序下有个 钉钉小程序，通常它们是“父子”关系，那可以基于支付宝小程序基础上兼容个性化即可，比如 [`@tarojs/plugin-platform-alipay-dd`](https://github.com/NervJS/taro-plugin-platform-alipay-dd)

具体内容可参加文档说明： https://docs.taro.zone/docs/platform-plugin/
